src/compiler/crystal/macros/methods.cr

```diff
@@ -1533,6 +1538,15 @@ module Crystal
       ArrayLiteral.new(defs)
     end
 
+    def self.has_method?(type, name)
+      type.defs.try &.each do |name, metadatas|
```

Comment:
You can simply use the `name` variable here, there's no need to traverse all metadatas